sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientLiveTests.cs

```diff
@@ -678,44 +678,92 @@ public async Task CanQueryWithFormattedQuery(FormattableStringWrapper query)
 
         [LiveOnly]
         [Test]
-        public async Task CanQueryResource()
+        public async Task CanQueryResourceGenericPrimaryWorkspace()
         {
+            var timespan = TimeSpan.FromSeconds(5);
+
             var client = CreateClient();
+            // Empty check
+            var results = await client.QueryResourceAsync<DateTimeOffset>(
+                new ResourceIdentifier(TestEnvironment.WorkspacePrimaryResourceId),
+                $"{_logsTestData.TableAName} | distinct * | project {LogsTestData.TimeGeneratedColumnName}",
+                timespan);
 
-            var results = await client.QueryResourceAsync(new ResourceIdentifier(TestEnvironment.StorageAccountId),
-                "search *",
-                _logsTestData.DataTimeRange);
+            Assert.AreEqual(0, results.Value.Count);
 
-            Assert.AreEqual(LogsQueryResultStatus.Success, results.Value.Status);
-            var resultTable = results.Value.Table;
-            CollectionAssert.IsNotEmpty(resultTable.Rows);
-            CollectionAssert.IsNotEmpty(resultTable.Columns);
+            // Check if all rows in table were uploaded
+            // Get the time of the third event and add a bit of buffer to it (events are 2d apart)
+            var maxOffset = (DateTimeOffset)_logsTestData.TableA[2][LogsTestData.TimeGeneratedColumnNameSent];
+            timespan = Recording.UtcNow - maxOffset;
+            timespan = timespan.Add(TimeSpan.FromDays(7));
 
-            bool verifyRow = false;
-            bool verifyColumn1 = false;
-            bool verifyColumn2 = false;
-            foreach (LogsTableRow rows in resultTable.Rows)
-            {
-                foreach (var row in rows)
-                {
-                    if ((row != null) && row.ToString().Contains("Create/Update Storage Account"))
-                    {
-                        verifyRow = true;
-                    }
-                }
-            }
+            // Make sure there is some data in the range specified
+            results = await client.QueryResourceAsync<DateTimeOffset>(
+                new ResourceIdentifier(TestEnvironment.WorkspacePrimaryResourceId),
+                $"{_logsTestData.TableAName} | distinct * | project {LogsTestData.TimeGeneratedColumnName}",
+                timespan);
 
-            foreach (LogsTableColumn columns in resultTable.Columns)
-            {
-                if (columns.Name == "SubscriptionId" && columns.Type == LogsColumnType.StringTypeValue)
-                    verifyColumn1 = true;
+            Assert.GreaterOrEqual(results.Value.Count, 3);
+        }
 
-                if (columns.Name == "TimeGenerated" && columns.Type == LogsColumnType.DatetimeTypeValue)
-                    verifyColumn2 = true;
-            }
+        [LiveOnly]
```

Comment:
Same question about whether this still needs to be LiveOnly.

Reply:
I agree - don't think they need to be [LiveOnly] anymore